plugins/github-actions/src/extensions/link.tsx

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2022 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from 'react';
+
+import { ComponentAdaptationOf } from '@backstage/core-plugin-api';
+import { linkComponentRef } from '@backstage/core-components';
+
+export const Provider: ComponentAdaptationOf<typeof linkComponentRef> = ({
```

Comment:
```suggestion
export const Adaptation: ComponentAdaptationOf<typeof linkComponentRef> = ({
```

What do you think about something like that?

On a separate note we could do something like the ApiRefs do and make `typeof linkComponentRef.T` possible?

Reply:
The `T` of a component ref can refer to multiple things, the Props, Context, Component itself or the Adaptation of it. We could have `.TProps`, `.TContext`, `.TComponent`, `.TAdaptation` so that you could:

```tsx
typeof linkComponentRef.TAdaptation
// ===
ComponentAdaptationOf<typeof linkComponentRef>
```

I'm not sure which is nicest, I kind of like `YadaOf<typeof yada>` as a design pattern. It's very common (in fact, in many languages). The inverted opaque non-value `.T` is perhaps shorter in syntax, and you don't need to know what the `*Of<>` wrapper is called (and you don't need to import it), and that's nice. But it's also a little backwards, and you need to fake/null the property when constructing the object. Do we have a preferred way of dealing with this in Backstage?

Reply:
Yep no this makes if there are multiple different types to extract